src/convertToObject.js

```diff
@@ -3,16 +3,60 @@
 /**
  * Implement convertToObject function:
  *
- * Function takes a string with styles (see an example in [stylesString.js](./stylesString.js))
+ * Function takes a string with styles
+ * (see an example in [stylesString.js](./stylesString.js))
  * and returns an object where CSS properties are keys
- * and values are the values of related CSS properties (see an exampl in [test file](./convertToObject.test.js))
+ * and values are the values of related CSS properties
+ * (see an exampl in [test file](./convertToObject.test.js))
  *
  * @param {string} sourceString
  *
  * @return {object}
  */
+// function convertToObject(sourceString) {
+//   const str = sourceString;
+//   let resultStr = '';
+//   const obj = {};
+
+//   for (const ch of str) {
+//     resultStr += ch.trim();
+//   }
+
+//   const arr = resultStr.split(';');
+
+//   arr.pop();
+
+//   const resultArray = arr.map(item => item.split(':'));
+
+//   const result = [].concat(...resultArray);
+
+//   for (let i = 0; i < result.length; i += 2) {
+//     if (result[i] === '') {
+//       continue;
+//     }
+//     obj[result[i]] = result[i + 1];
+//   }
+
+//   return obj;
+// }
+
 function convertToObject(sourceString) {
-  // write your code here
+  const obj = {};
+  const array = sourceString.split(';').map(item => item.split(':'));
+
+  const result = [].concat(...array);
```

Comment:
Better is to do this with the spread operator.
`const result = [...array];`

[More info](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax)

Reply:
The code doesn't work when I do so

Reply:
![image](https://user-images.githubusercontent.com/56752562/219330276-552e6a6d-bbcd-479b-a9ce-6dd56cf5219c.png)

Reply:
The same result